packages/orchestrator-ui-components/package.json

```diff
@@ -23,7 +23,7 @@
     ],
     "scripts": {
         "test": "jest",
-        "build": "tsup src/index.ts --format esm --dts --tsconfig tsconfig.build.json",
+        "build": "node -p \"'export const ORCHESTRATOR_UI_LIBRARY_VERSION = ' + JSON.stringify(require('./package.json').version) + ';'\" > src/configuration/version.ts && prettier --write src/configuration/version.ts && tsup src/index.ts --format esm --dts --tsconfig tsconfig.build.json",
```

Comment:
This will work. I would just make the version file geneation it's own step and then call that step in the build step.